src/main/java/com/google/googleidentity/oauth2/authorizationcode/CodeStore.java

```diff
@@ -0,0 +1,43 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.authorizationcode;
+
+import com.google.googleidentity.oauth2.request.OAuth2Request;
+
+import java.util.Optional;
+
+/**
+ * AuthorizationCode store interface
+ */
+public interface CodeStore {
+
+    /**
+     *
+     * @param code
+     * @return the related request and delete the code
+     */
+    Optional<OAuth2Request> consumeCode(String code);
+
+    /**
+     * set the code for the request
+     *
+     * @param code
+     * @param request
+     * @return success or not
```

Comment:
First, it's not explained when this might fail.

But, codes are random. So we can always generate a new one. Maybe pushing code generation down into CodeStore means this can always succeed, since we can retry generating a code. Otherwise, since failure should be very rare, it might be good to signal failure here via an exception rather than a boolean return value.